### ISSUE TITLE
Is PointNetClassifier implemented according to the original (cited) paper?

### ISSUE THREAD

**Issue by ngamulin** [NVIDIAGameWorks/kaolin#186]
Hi, 

I have tried to draw the architecture of the model by hand according to the article, and then compared the original author's implementation in [TensorFlow](https://github.com/charlesq34/pointnet) and PointNetClassifier in this repo. I might be wrong, but it seems that the PointNetClassifier is equivalent to Input transform. Below is the copy of TensorFlow implementation:
`def input_transform_net(point_cloud, is_training, bn_decay=None, K=3):
    """ Input (XYZ) Transform Net, input is BxNx3 gray image
        Return:
            Transformation matrix of size 3xK """
    batch_size = point_cloud.get_shape()[0].value
    num_point = point_cloud.get_shape()[1].value

    input_image = tf.expand_dims(point_cloud, -1)
    net = tf_util.conv2d(input_image, 64, [1,3],
                         padding='VALID', stride=[1,1],
                         bn=True, is_training=is_training,
                         scope='tconv1', bn_decay=bn_decay)
    net = tf_util.conv2d(net, 128, [1,1],
                         padding='VALID', stride=[1,1],
                         bn=True, is_training=is_training,
                         scope='tconv2', bn_decay=bn_decay)
    net = tf_util.conv2d(net, 1024, [1,1],
                         padding='VALID', stride=[1,1],
                         bn=True, is_training=is_training,
                         scope='tconv3', bn_decay=bn_decay)
    net = tf_util.max_pool2d(net, [num_point,1],
                             padding='VALID', scope='tmaxpool')

    net = tf.reshape(net, [batch_size, -1])
    net = tf_util.fully_connected(net, 512, bn=True, is_training=is_training,
                                  scope='tfc1', bn_decay=bn_decay)
    net = tf_util.fully_connected(net, 256, bn=True, is_training=is_training,
                                  scope='tfc2', bn_decay=bn_decay)

    with tf.variable_scope('transform_XYZ') as sc:
        assert(K==3)
        weights = tf.get_variable('weights', [256, 3*K],
                                  initializer=tf.constant_initializer(0.0),
                                  dtype=tf.float32)
        biases = tf.get_variable('biases', [3*K],
                                 initializer=tf.constant_initializer(0.0),
                                 dtype=tf.float32)
        biases += tf.constant([1,0,0,0,1,0,0,0,1], dtype=tf.float32)
        transform = tf.matmul(net, weights)
        transform = tf.nn.bias_add(transform, biases)

    transform = tf.reshape(transform, [batch_size, 3, K])
    return transform`

This is the text from the article:

> The first transformation network is a mini-PointNet that takes raw point cloud as input and regresses to a 3 × 3 matrix. It’s composed of a shared MLP(64, 128, 1024) network (with layer output sizes 64, 128, 1024) on each point, a max pooling across points and two fully connected layers with output sizes 512, 256. The output matrix is initialized as an identity matrix. All layers, except the last one, include ReLU and batch normalization. The second transformation network has the same architecture as the first one except that the output is a 64 × 64 matrix. The matrix is also initialized as an identity. A regularization loss (with weight 0.001) is added to the softmax classification loss to make the matrix close to orthogonal. We use dropout with keep ratio 0.7 on the last fully connected layer, whose output dimension 256, before class
score prediction. The decay rate for batch normalization starts with 0.5 and is gradually increased to 0.99. We use adam optimizer with nitial learning rate 0.001, momentum 0.9 and batch size 32.

I might be wrong but I would be thankful if anyone clarified this. It seems that the second part is missing.

**Comment by pushkalkatara** [comment]
Hi @ngamulin By second part do you mean the feature processing net before the classification layer?

I think the network is correct as feature extraction is done through [PointNetFeatureExtractor](https://github.com/NVIDIAGameWorks/kaolin/blob/be02fa3c621682da713e41426e65f63c7531a4ff/kaolin/models/PointNet.py#L292) with `layer_dims=feat_layer_dims` and `feat_layer_dims: Iterable[int] = [64, 128]` in the `PointNetClassifier`. 64 being the input processing dim and 128 being the feature processing dim.

**Comment by Caenorst** [comment]
Closing for inactivity as we dropped full model implementation since v0.9
